### PR TITLE
Fix stack underflows when yielding twice

### LINKED ISSUE #30269: GDScript stack underflow while yielding to signals (regression from #28884)
**Godot version:**
3.2-dev 4d99408d123e4eea5773c8d1e35b59385b0aefa1

Regression from #28884.

**OS/device including version:**
Windows 10

**Issue description:**
While running GUT tests encountered this:

```
Debugger Break, Reason: 'Stack Underflow (Engine Bug)'
ERROR: GDScriptLanguage::debug_get_stack_level_function: Index p_level=0 out of size (_debug_call_stack_pos=0)
   At: modules\gdscript\gdscript_editor.cpp:271
ERROR: GDScriptLanguage::debug_get_stack_level_line: Index p_level=0 out of size (_debug_call_stack_pos=0)
   At: modules\gdscript\gdscript_editor.cpp:260
ERROR: GDScriptLanguage::debug_get_stack_level_source: Index p_level=0 out of size (_debug_call_stack_pos=0)
   At: modules\gdscript\gdscript_editor.cpp:280
*Frame 0 - :-1 in function ''
Enter "help" for assistance.
debug>
```

**Steps to reproduce:**
```
git clone https://github.com/bitwes/Gut/
```
Now open the project, run main scene (default). Godot Test Window shall appear.
Select `res://test/unit/test_gut_yielding.gd` from the popup menu [`...`] button:

Press `Run`. Stalls on the first yield test (`test_can_yield_using_built_in_timer`).

### PR DESCRIPTION
**Pull request by bojidar-bg**
Also, refactor `GDScriptFunctionState::_signal_callback`, removing some excessive repetition.
Fixes #30269.

### CODE REVIEW

File: modules/gdscript/gdscript_function.cpp [hunk 1]
```diff
@@ -1823,44 +1812,7 @@ Variant GDScriptFunctionState::_signal_callback(const Variant **p_args, int p_ar
...
-	function = NULL; //cleaned up;
-	state.result = Variant();
-
-	if (completed) {
-		if (first_state.is_valid()) {
-			first_state->emit_signal("completed", ret);
-		} else {
-			emit_signal("completed", ret);
-		}
-	}
```
Comment (bojidar-bg) [inline modules/gdscript/gdscript_function.cpp]: Note that this closing brace is after the `#ifdef` in `resume`, but before it here.
Making it the same as in `resume` fixes the issue already... but who wants a -1+1 PR when you can get a -49+1?

**Review by Xrayez — APPROVED** [review]
All my GUT tests pass now, thanks. 😃

**Comment by owstetra** [comment]
@bojidar-bg REALLY THANK YOU VERY VERY MUCH !!!
this bug was stopping us from moving further in the Project !!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**